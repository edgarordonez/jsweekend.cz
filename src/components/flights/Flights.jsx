import React, { Fragment } from 'react';
import { graphql } from 'react-apollo';
import SearchFlightsQuery from './searchFlightsQuery';

export const Flights = ({data: { allFlights, loading, refetch }}) => {
  return (
    <Fragment>
      {
        loading ?
          <div>
            <p>Loading...</p>
          </div>
        :
          <div>
            {JSON.stringify(allFlights, null, 2)}
          </div>
      }
    </Fragment>
  );
}

const FlightsWrapper = graphql(SearchFlightsQuery, {
  options: ({ fields: { from , to, date } }) => ({ variables: { from, to, date }}),
})(Flights);

export default FlightsWrapper;
