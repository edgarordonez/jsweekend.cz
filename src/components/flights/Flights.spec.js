import React from 'react';
import { shallow } from 'enzyme';
import { Flights } from './Flights';

it('renders without crashing', () => {
  const data = {
    allFlights: {
      edges: [
        {
          "node": {
            "id": "RmxpZ2h0OjMzNDEwNjkwMw==",
            "airlines": [
              {
                "name": "Vueling",
                "code": "VY",
                "logoUrl": "https://images.kiwi.com/airlines/64/VY.png"
              }
            ],
            "departure": {
              "airport": {
                "locationId": "BCN",
                "name": "Aeropuerto de Barcelona-El Prat",
                "slug": "aeropuerto-de-barcelona-el-prat-barcelona-espana",
                "city": {
                  "name": "Barcelona"
                }
              },
              "time": "2018-02-23T17:10:00.000Z",
              "localTime": "2018-02-23T18:10:00.000Z"
            },
            "arrival": {
              "airport": {
                "locationId": "LTN",
                "name": "Aeropuerto de Londres-Luton",
                "slug": "aeropuerto-de-londres-luton-londres-reino-unido",
                "city": {
                  "name": "Londres"
                }
              }
            }
          }
        }
      ]
    },
    loading: true,
    loadMoreFlights: (event) => event
  }

  const flights = shallow(<Flights allFlights={data.allFlights} loading={data.loading} loadMoreFlights={data.loadMoreFlights} />);

  expect(flights).toMatchSnapshot();
});
