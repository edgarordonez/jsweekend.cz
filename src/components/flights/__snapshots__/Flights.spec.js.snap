// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    data={
        Object {
            "allFlights": Object {
              "edges": Array [
                Object {
                  "node": Object {
                    "airlines": Array [
                      Object {
                        "code": "VY",
                        "logoUrl": "https://images.kiwi.com/airlines/64/VY.png",
                        "name": "Vueling",
                      },
                    ],
                    "arrival": Object {
                      "airport": Object {
                        "city": Object {
                          "name": "Londres",
                        },
                        "locationId": "LTN",
                        "name": "Aeropuerto de Londres-Luton",
                        "slug": "aeropuerto-de-londres-luton-londres-reino-unido",
                      },
                    },
                    "departure": Object {
                      "airport": Object {
                        "city": Object {
                          "name": "Barcelona",
                        },
                        "locationId": "BCN",
                        "name": "Aeropuerto de Barcelona-El Prat",
                        "slug": "aeropuerto-de-barcelona-el-prat-barcelona-espana",
                      },
                      "localTime": "2018-02-23T18:10:00.000Z",
                      "time": "2018-02-23T17:10:00.000Z",
                    },
                    "id": "RmxpZ2h0OjMzNDEwNjkwMw==",
                  },
                },
              ],
            },
            "loading": true,
            "refetch": [Function],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <div>
        <p>
                Loading...
        </p>
</div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <p>
          Loading...
</p>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Loading...",
        },
        "ref": null,
        "rendered": "Loading...",
        "type": "p",
      },
      "type": "div",
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <div>
          <p>
                    Loading...
          </p>
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <p>
            Loading...
</p>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Loading...",
          },
          "ref": null,
          "rendered": "Loading...",
          "type": "p",
        },
        "type": "div",
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
